### DISCUSSION TITLE
Hi, I cannot export my CodePen work out as a Gist to GitHub

### DISCUSSION THREAD

**Discussion by HafisMamoun** [community/community#63888] [New to GitHub]
Hi, I am new to GitHub, and I cannot export my work from Codepen as a Gist onto GitHub it says Authentication Has Expired, how do I reauthenticate my GitHub account to export my work from Codepen on to GitHub as a Gist.

Thank you 

H. Mamoun

**Comment by oftharsh** [comment]
If you're encountering an "Authentication Has Expired" error when trying to export your work from CodePen to GitHub as a Gist, it's likely that the authentication token between CodePen and GitHub needs to be refreshed. Here's how you can reauthenticate your GitHub account on CodePen:

1. **Sign Out and Sign In Again**:
   - Log out of your CodePen account (if you're already logged in).
   - Clear your browser's cache and cookies to ensure a clean session.
   - Log back in to your CodePen account.

2. **Reconnect GitHub Account**:
   - Once you're logged in to CodePen, go to your account settings or profile.
   - Look for the GitHub integration settings or connected accounts section.
   - Disconnect your GitHub account if it's already connected.
   - Reconnect your GitHub account by following the steps provided by CodePen.

3. **Generate a New Personal Access Token**:
   - If the reconnection doesn't solve the issue, you might need to generate a new Personal Access Token on GitHub.
   - Go to your GitHub account settings.
   - Under "Developer settings," select "Personal access tokens."
   - Generate a new token with the necessary scopes for Gist (e.g., "gist" scope).
   - Copy the newly generated token.

4. **Update Token on CodePen**:
   - Go back to your CodePen account settings or profile.
   - Find the GitHub integration settings or connected accounts section.
   - Update the Personal Access Token with the newly generated token from GitHub.

5. **Export to Gist**:
   - Once your GitHub account is reconnected and the token is updated, try exporting your work from CodePen to GitHub as a Gist again.

If you're still facing issues after following these steps, you might want to contact CodePen's support or community forums for further assistance. They can provide more specific guidance based on their platform's current setup and any recent changes.

**Comment by HafisMamoun** [comment]
Thank you

**Comment by Jaber-Saed** [comment]
If you're encountering an "Authentication Has Expired" error when trying to export your work from CodePen to GitHub as a Gist, it's likely that your authentication token has expired. To resolve this, you'll need to reauthenticate your GitHub account with CodePen. Here's what you can do:

    Log In to CodePen: Make sure you are logged in to your CodePen account.

    Open Your Profile Settings: Click on your profile picture or username in the top-right corner of the CodePen interface to access your account settings.

    Access GitHub Integration: Look for a section in your profile settings related to GitHub integration, authorization, or connected accounts.

    Disconnect GitHub: You may see an option to disconnect your GitHub account from CodePen. This is usually done to revoke CodePen's access to your GitHub account. Click on this option to disconnect.

    Reconnect GitHub: After disconnecting, you should now be able to reconnect your GitHub account. This usually involves clicking a button that says "Connect GitHub" or similar.

    Follow the Authorization Steps: When you click to reconnect, CodePen will prompt you to authorize access to your GitHub account. This might involve logging in to GitHub and granting permissions to CodePen.

    Complete the Integration: Once you've authorized CodePen, your GitHub account should be successfully reconnected.

    Try Exporting Again: Now that your GitHub account is reconnected, try exporting your work as a Gist again. This time, the authentication should not be expired.

If you're still facing issues after reauthenticating, you might want to check if there are any issues with your GitHub account itself, such as revoked access, expired tokens, or any security settings that might be affecting the connection.

If the problem persists, it could be helpful to reach out to CodePen's support or community forums for further assistance, as they would have specific insights into their integration process and any known issues.

**Comment by prajasus2308** [comment]
Go to your Pen in CodePen.

At the bottom of the editor, click Export → Export as Gist.

You’ll be prompted to connect your GitHub account (via OAuth).

Once connected, CodePen will create a Gist in your GitHub profile containing your Pen’s code.

Limitation: Gists are not full repositories; they’re snippets. You can’t push updates automatically from CodePen to a repo.
